streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1081,8 +1081,9 @@ private void shutdown(final boolean cleanRun) {
         streamsMetrics.removeAllSensors();
     }
 
+    // visible for testing
     @SuppressWarnings("ThrowableNotThrown")
-    private void shutdownTasksAndState(final boolean cleanRun) {
+    void shutdownTasksAndState(final boolean cleanRun) {
```

Comment:
@guozhangwang i completely agree with your comment above about "testing a class' out-facing behavior". That is exactly what we should be testing, as soon as we start having to expose methods as package-private etc for testing, then we really need to look at the design. So, the same thing applies for testing with reflection IMO. It is the same issue, presented in a different way. The reason i don't like it is that the reflection adds a bunch of noise to the tests making them harder to follow and is digging into the innards of the class we are testing. So, for me, the lesser or two evils would be to expose the method as package-private (begrudgingly!).

But yes, calling `close` seems fine in this case